actions/parse_swarmbench_output.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/python3
+
+import sys
+import subprocess
+from charms.docker import Docker
+from charms.benchmark import Benchmark
+
+
+def action_set(key, val):
+    ''' Helper method to set key/val for benchmark data '''
+    action_cmd = ['action-set']
+    if isinstance(val, dict):
+        for k, v in val.items():
+            action_set('{}.{}'.format(key, k), v)
+        return
+
+    action_cmd.append('{}={}'.format(key, val))
+    subprocess.check_call(action_cmd)
+
+
+def parse_output(container_id):
+    d = Docker()
+    raw_logs = d.logs(container_id)
+
+    run_output = raw_logs.splitlines()
+
+    # The final 2 lines we care about have some specific strings
+    # search for them and determine if it was actually a successful run
```

Comment:
I don't see any if checks here for detectable errors, such as `raw_logs` being '' or None (then you probably shouldn't `splitlines`). Normal parsers will throw exceptions with line and column numbers but I could envision a scenario where the string manipulation could fail and the user/operator will not know where to look. Since I don't know this feature very well I am willing to let robustness be a future enhancement.